tracer/src/Datadog.Trace/FeatureFlags/Exposure/ExposureApi.cs

```diff
@@ -0,0 +1,163 @@
+            tracerSettings.Manager.SubscribeToChanges(settingChanges => UpdateSettings(settingChanges.UpdatedExporter ?? settingChanges.PreviousExporter, settingChanges.UpdatedMutable ?? settingChanges.PreviousMutable));
+        }
+
+        private void UpdateSettings(ExporterSettings exporterSettings, MutableSettings settings)
+        {
+            Log.Debug("ExposureApi::UpdateSettings -> Applyting settings");
+            var apiRequestFactory = AgentTransportStrategy.Get(
+                exporterSettings,
+                productName: "FeatureFlags exposure",
+                tcpTimeout: TimeSpan.FromSeconds(5),
+                AgentHttpHeaderNames.MinimalHeaders,
+                () => new MinimalAgentHeaderHelper(),
+                uri => uri);
+            Interlocked.Exchange(ref _apiRequestFactory, apiRequestFactory);
+
+            var context = new Dictionary<string, string>
+            {
+                { "service", settings.ServiceName ?? "unknown" },
+                { "env", settings.Environment ?? "unknown" },
+                { "version", settings.ServiceVersion ?? "unknown" }
+            };
+            Interlocked.Exchange(ref _context, context);
+        }
+
+        public static ExposureApi Create(TracerSettings tracerSettings)
+        {
+            return new ExposureApi(tracerSettings);
+        }
+
+        public void Dispose()
+        {
+            try { _processExit.TrySetResult(true); }
+            catch { }
+        }
+
+        public void TryToStartSendLoopIfNotStarted()
+        {
+            if (_started != 0 || Interlocked.Exchange(ref _started, 1) != 0)
+            {
+                return;
+            }
+
+            _ = Task.Run(SendLoopAsync).ContinueWith(t => { Log.Error(t.Exception, "FeatureFlags Exposure send loop failed"); }, TaskContinuationOptions.OnlyOnFaulted);
+        }
+
+        private async Task SendLoopAsync()
+        {
+            Log.Debug("ExposureApi::SendLoopAsync -> Enter");
+            while (!_processExit.Task.IsCompleted)
+            {
+                try
+                {
+                    var apiRequestFactory = _apiRequestFactory;
+                    if (apiRequestFactory is not null)
+                    {
+                        var uri = apiRequestFactory.GetEndpoint($"evp_proxy/v2/{ExposurePath}");
+                        if (_exposures.Count > 0)
+                        {
+                            await SendBatchAsync(apiRequestFactory, uri).ConfigureAwait(false);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Error while sending Feature Flags exposures to the agent");
+                }
+
+                try
+                {
+                    await Task.WhenAny(_processExit.Task, Task.Delay(_sendInterval)).ConfigureAwait(false);
```

Comment:
This will have some minor drift depending on how long the loop takes to run - I assume that's fine?